pkg/markers/markers.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package markers
+
+import "fmt"
+
+const (
+	// OptionalMarker is the marker that indicates that a field is optional.
+	OptionalMarker = "optional"
+
+	// RequiredMarker is the marker that indicates that a field is required.
+	RequiredMarker = "required"
+)
+
+const (
+	// KubebuilderRootMarker is the marker that indicates that a struct is the object root for code and CRD generation.
+	KubebuilderRootMarker = "kubebuilder:object:root"
+
+	// KubebuilderStatusSubresourceMarker is the marker that indicates that the CRD generated for a struct should include the /status subresource.
+	KubebuilderStatusSubresourceMarker = "kubebuilder:subresource:status"
+
+	// KubebuilderEnumMarker is the marker that indicates that a field has an enum in kubebuilder.
+	KubebuilderEnumMarker = "kubebuilder:validation:Enum"
+
+	// KubebuilderFormatMarker is the marker that indicates that a field has a format in kubebuilder.
+	KubebuilderFormatMarker = "kubebuilder:validation:Format"
+
+	// KubebuilderMaxItemsMarker is the marker that indicates that a field has a maximum number of items in kubebuilder.
+	KubebuilderMaxItemsMarker = "kubebuilder:validation:MaxItems"
+
+	// KubebuilderMaxLengthMarker is the marker that indicates that a field has a maximum length in kubebuilder.
+	KubebuilderMaxLengthMarker = "kubebuilder:validation:MaxLength"
+
+	// KubebuilderOptionalMarker is the marker that indicates that a field is optional in kubebuilder.
+	KubebuilderOptionalMarker = "kubebuilder:validation:Optional"
+
+	// KubebuilderRequiredMarker is the marker that indicates that a field is required in kubebuilder.
+	KubebuilderRequiredMarker = "kubebuilder:validation:Required"
+
+	// KubebuilderItemsMaxLengthMarker is the marker that indicates that a field has a maximum length in kubebuilder.
+	KubebuilderItemsMaxLengthMarker = "kubebuilder:validation:items:MaxLength"
+
+	// KubebuilderItemsEnumMarker is the marker that indicates that a field has an enum in kubebuilder.
+	KubebuilderItemsEnumMarker = "kubebuilder:validation:items:Enum"
+
+	// KubebuilderItemsFormatMarker is the marker that indicates that a field has a format in kubebuilder.
+	KubebuilderItemsFormatMarker = "kubebuilder:validation:items:Format"
+)
+
+func Format(marker, value string) string {
+	return fmt.Sprintf("%s:=%s", marker, value)
```

Comment:
Is this format always correct? What about markers that have multiple parameters like `kubebuilder:validation:XValidation`?

Reply:
No, it is not